### ISSUE TITLE
Add support for CNTK

### ISSUE THREAD

**Issue by cesarsouza** [cesarsouza/keras-sharp#10]
While TensorFlow gradient's issue is pending, we can start adding support for CNTK to exercise the frontend.

From https://github.com/accord-net/framework/issues/904

**Comment by cesarsouza** [comment]
Backends have been added for CNTK. However, a few issues on CNTKs side currently prevent full models from being eventually learned. For instance, using the CNTK backend might result in multiple [AccessViolationExceptions](https://msdn.microsoft.com/en-us/library/system.accessviolationexception(v=vs.110).aspx) which are not present when using the TensorFlowSharp backend, and have thus nothing to do with the main engine (since the engine does not use any unmanaged code at all at this moment).
